File: src/app/core/models/commande.ts
export interface LigneCommande {
  platId: number;
  nomPlat: string;
  quantite: number;
  prixUnitaire: number;
}

export interface Commande {
  id: number;
  clientId: number;
  restaurantId: number;
  lignes: LigneCommande[];
  statut: 'en_attente' | 'en_preparation' | 'livree';
  dateCommande: string;
  adresseLivraison: string;
}

File: backend/business-service/src/main/java/com/foodrush/business/model/Commande.java
package com.foodrush.business.model;

import jakarta.persistence.*;
import lombok.Data;
import java.util.List;

@Entity
@Table(name = "commandes")
@Data
public class Commande {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "client_id")
    private String clientId;
    
    @Column(name = "restaurant_id")
    private Long restaurantId;
    
    private String statut = "en_attente";
    
    @Column(name = "date_commande")
    private String dateCommande;
    
    @Column(name = "adresse_livraison")
    private String adresseLivraison;
    
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "commande_id")
    private List<LigneCommande> lignes;
}
